### ISSUE TITLE
(MkII head) Looper pedal + hidden-amp compatibility

### ISSUE THREAD

**Issue by 7043mcgeep** [katana-dev/docs#9]
Hello,

I know this github has been inactive for a few years, but I figured this issue should go here, since I've read that @gumtown has been working on the port of FloorBoard to the MKII.

I use a Boss RC-1 looper pedal, plugged straight into my KTN-HEAD MKII. I downloaded the live set containing the hidden pre-amps types (only 15 settings show up in Tone Studio). I recorded something on my looper using a standard (1-5 amp type) setting. Then, with the track looping, I loaded one of the hidden amp settings. Doing this stopped my looper pedal and erased the track that was looping. I found that I was unable to use my looper pedal with any of the hidden amp live sets loaded (no sound coming out of the looper), which seems like a very strange limitation and left me with a few questions.

What _are_ these hidden amp settings? (because they actually are pretty darn good and make me wish they were officially offered).

How on earth did you find them sitting in memory? What does it have to do with the GT-100? How did you make it function as a live set?

I was expecting that none of these hidden amps would work on the MKII, as I've seen many (all?) MKI settings are not compatible with the MKII. But 15 are actually loadable for me, just with the strange problem with the send/return through the looper. My best guess is that there's an updated memory system on MKII and only half of the hidden amps are successfully retrieved from valid location. Either that, or they removed some hidden amps to make room for more functionality. Who knows.

I like this repository. Was not expecting to find something as interesting as "hidden amp models" for this thing. I imagine it makes for some fun projects.
Ping @Beanow

**Comment by gumtown** [comment]
Hi,
Yes I am still working on a MKii version of FxFloorBoard, it has stalled for a bit as I don't have a mkii, so I will have to call on other mk2 users for help testing things.
The mk2 sneaky amps live set is something that I put together while designing the FxFloorBoard mk2 file conversion to Mk1, and posted to the Admin on a Facebook Katana page to try out, most likely the S/R Loop is switched 'off' in each patch, if you enable the loop when trying the 'Sneaky Amp" you will hear it through your looper.
There are actually 2 sneaky mk2 amp files, as the mk2 tsl patch bundle can only contain a maximum of 15 patches. There should be a "Sneaky amps for mk2 1~15" and a "Sneaky amps for mk2 16~30" files.
The Katana internal structure is based on the GT-100 subset of features, when the first *.tsl patch file became available from tone studio, I disassembled it and found references to "GT-100" as the device type. 
I then modified the GT-100FxFloorBoard editor with, the Katana system exclusive device ID identifier, and disabled file size checks, and released it as an Alpha test unit to see what happened. Results came back that all the preamps (which I left all GT-100 effects intact) and many of the effects found in the GT-100 seemed to work in the Katana.
The MK2 now has a completely different memory architecture, only 1/2 the size of the mk1, and many sneaky effects, such as tera-echo and over-tone are not there.

Hope that enlightens you :-)

Cheers
Colin.

**Comment by Beanow** [comment]
Yeah as @gumtown mentioned. I have a Katana-100 from the first lineup.

@snhirsch's initial [reverse engineering](https://github.com/snhirsch/katana-midi-bridge/blob/master/doc/katana_sysex.txt) got me set up to send sysex requests to the Katana mk1. Discussion with them and on https://www.vguitarforums.com led to the idea it's very similar to a GT-100.
@gumtown's FxFloorBoard exposed all of the GT-100 features and the GT-100 documentation provided a lot of reference. From there I spent a good few hours testing everything to fill out the details of these documents.

I've not worked with, or read up on the mk2 internals at all personally.

> There should be a "Sneaky amps for mk2 115" and a "Sneaky amps for mk2 1630" files.

@gumtown if you or others are interested in adding to these repositories let me know. Happy to take pull requests and/or maintainers.

**Comment by skconrad** [comment]
Latest Katana MK2 knowledge.

https://www.vguitarforums.com/smf/index.php?board=317.0

**Comment by 7043mcgeep** [comment]
Thanks for all the helpful info, everyone.

> Loop is switched 'off' in each patch, if you enable the loop when trying the 'Sneaky Amp" you will hear it through your looper.

Good to know! Best of luck on the project.

**Comment by gumtown** [comment]
Update !! 
KAtanaFxFloorBoard editor for Mk1 and Mk2 have now had a revamp and upgrade and are available at
https://sourceforge.net/projects/fxfloorboard/files/KatanaFxFloorBoard/
